sdk/src/metrics/aggregation/lastvalue_aggregation.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef ENABLE_METRICS_PREVIEW
+#  include "opentelemetry/sdk/metrics/aggregation/lastvalue_aggregation.h"
+#  include "opentelemetry/common/timestamp.h"
+#  include "opentelemetry/version.h"
+
+#  include <mutex>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+
+LongLastValueAggregation::LongLastValueAggregation() : is_lastvalue_valid_(false) {}
+
+void LongLastValueAggregation::Aggregate(long value, const PointAttributes &attributes) noexcept
+{
+  const std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
+  is_lastvalue_valid_ = true;
+  value_              = value;
+}
+
+PointType LongLastValueAggregation::Collect() noexcept
+{
+  if (!is_lastvalue_valid_)
```

Comment:
Don't we need to protect `is_lastvalue_valid_` here?

Reply:
Good point. I thought it should be fine to return an invalid value in given Collect() call, as long as the subsequent call would return a valid value. Worst case, we may miss a given metric in a series of Collect() calls, but this way we can avoid a lock.

Reply:
I'm afraid this would be a UB, and the lock for reading is unavoidable.

Reply:
I thought UB should be fine as long as  `is_lastvalue_valid_` returns either true/false (was not expecting strong consistency here). As in one of the next few iterations, it would be true eventually. But have added the lock if it's causing confusion.